inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -87,17 +87,31 @@ bool concat_in_place_optimization::match(concatenation_node& node) {
     if (node.has_fused_primitives() || !node.get_fused_activations_funcs().empty())
         return false;
 
-    // For in place concatenation input layouts and data types must match.
-    auto output_format = node.get_output_layout().format;
-    auto output_datatype = node.get_output_layout().data_type;
-    auto concat_axis = node.get_primitive()->axis;
+    bool is_onednn_impl = false;
 
-    // oneDNN doens't support paddings and such concat optimizations
     for (auto& input : node.get_dependencies()) {
-        if (input->get_preferred_impl_type() == impl_types::onednn)
+        if (input->get_preferred_impl_type() == impl_types::onednn) {
+            is_onednn_impl = true;
+            break;
+        }
+    }
+
+    // oneDNN support paddings and such concat optimizations Only use_usm and batch 1.
```

Comment:
What about: `// Implicit concat for onednn only when use_usm and batch 1.`

Reply:
Agree, updated it.